qiskit_rigetti/_qcs_job.py

```diff
@@ -107,6 +109,28 @@ def _start_circuit(self, circuit: QuantumCircuit) -> Response:
         # typing: QuantumComputer's inner QAM is generic, so we set the expected type here
         return cast(Response, self._qc.qam.execute(executable))
 
+    def _handle_barriers(self, qasm: str) -> str:
+        lines = []
+        for line in qasm.splitlines():
+            if not line.startswith("barrier"):
+                lines.append(line)
+                continue
+
+            qubits = line[(len("barrier ")) :]
+            num_qubits = qubits.count(",") + 1
+            if num_qubits != self._configuration.num_qubits:
```

Comment:
Just checking: this doesn't require that the barrier uses all qubits on the *device*, just those used in the *circuit*?

Reply:
That's a good point. Thanks for noting that. Will adjust.

Reply:
OK, addressed in latest push